fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt

```diff
@@ -80,24 +80,41 @@ enum class DrawerValue {
  * State of the [Drawer] composable.
  *
  * @param initialValue The initial value of the state.
+ * @param expandable defines if the drawer is allowed to take the Expanded state.
+ * @param isSkipOpenState defines if the drawer is allowed to take the Open state. (Open State is skipped in case of true)
  * @param confirmStateChange Optional callback invoked to confirm or veto a pending state change.
  */
 class DrawerState(
     private val initialValue: DrawerValue = DrawerValue.Closed,
+    internal val expandable: Boolean = true,
+    internal val isSkipOpenState: Boolean = false,
     confirmStateChange: (DrawerValue) -> Boolean = { true }
 ) : SwipeableState<DrawerValue>(
     initialValue = initialValue,
     animationSpec = AnimationSpec,
     confirmStateChange = confirmStateChange
 ) {
-
+    init {
+        if (isSkipOpenState) {
+            require(initialValue != DrawerValue.Open) {
+                "The initial value must not be set to Open if skipOpenState is set to" +
+                        " true."
+            }
+        }
+        else if (expandable) {
```

Comment:
remove the else. check for expandable is independent check

Reply:
Done.